specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2017-04-01-preview/inputs.json

```diff
@@ -660,6 +676,35 @@
         }
       }
     },
+    "RawReferenceInputDataSource": {
+      "description": "Describes a raw input data source. This data source type is only applicable/usable when using the query testing API. You cannot create a job with this data source type or add an input/output of this data source type to an existing job.",
+      "x-ms-discriminator-value": "Raw",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ReferenceInputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RawInputDatasourceProperties",
```

Comment:
Even though we just re-use, RawInputDatasourceProperties, I think we created specific Stream and Reference properties definitions just in case they are extended even if they just reference the "base" at the time. See BlobReferenceInputDataSourceProperties as an example.

Is it worth doing the same here? Not sure if it is likely that there would be a difference between stream and reference in terms of properties.